### PR TITLE
Frontend Products, Categories, Filters

### PR DESCRIPTION
**Pull request by arijit1111**
Home page brand listing, testimonials, other sections
category pages with product listing

### CODE REVIEW

**Review by smaqeelkazmi — APPROVED** [review]

File: Modules/Shop/Routes/api.php [hunk 1]
```diff
@@ -13,13 +13,25 @@
...
 |
 */
+Route::middleware('auth:api')
+    ->prefix('shop')
+    ->name('shop.')
+    ->group(function () {
+        Route::get('/product', 'ShopController@fetchProductDetail')->name('product_detail');
+    });
 
 Route::prefix('shop')
```
Comment (smaqeelkazmi) [inline Modules/Shop/Routes/api.php:23]: @arijit1111 Where we are using these routes and why these aren't protected?

Comment (smaqeelkazmi) [reply Modules/Shop/Routes/api.php:23]: @arijit1111 Follow up

File: Modules/Shop/Http/Services/ShopService.php [hunk 2]
```diff
@@ -629,4 +771,238 @@ public function getProduct(int $productId): array
...
+
+        $results = Product::leftJoin('categories as main', 'products.main_category', '=', 'main.id')
+            ->leftJoin('categories as sub', 'products.category', '=', 'sub.id')
+            ->leftJoin('categories as child', 'products.sub_category', '=', 'child.id')
+            ->where('products.status', 'publish')
+            ->where('products.name', 'like', '%'.$request->search.'%')
+            ->orWhere('main.title', 'like', '%'.$request->search.'%' )
+            ->orWhere('sub.title', 'like', '%'.$request->search.'%' )
+            ->orWhere('child.title', 'like', '%'.$request->search.'%' )
+            ->get();;
```
Comment (smaqeelkazmi) [inline Modules/Shop/Http/Services/ShopService.php]: @arijit1111 If you need only one column as `name` you can use the `select` method to select specific columns.

Comment (arijit1111) [reply Modules/Shop/Http/Services/ShopService.php]: fixed

**Review by smaqeelkazmi — APPROVED** [review]
Approved

**Pull request merged**